lib/src/json_schema/utils.dart

```diff
@@ -106,3 +106,17 @@ class DefaultValidators {
     }
   }
 }
+
+extension DeepMerge<K> on Map<K, dynamic> {
+  /// Merge other into this with properties in this taking precedence.
+  Map<K, dynamic> deepMerge(Map<K, dynamic> other) {
+    other.forEach((key, value) {
+      if (this.containsKey(key) && value is Map && this[key].runtimeType == value.runtimeType) {
+        this[key] = (this[key] as Map).deepMerge(value);
+      } else {
+        this[key] = value;
```

Comment:
If the other map's key is of a different type than ours, does this clobber ours?

Reply:
Line 117 is clobbering the key in `this`. I meant to skip keys that already existed, unless the value for the key was a map.

This might also not be the right thing to be doing with the json-schema spec.